### PR TITLE
Cosmetic CLI output fixes

### PR DESCRIPTION
**Pull request by corylanou**
This fixes the following types of queries:

**FROM**:

```sql
> select * from cpu
name    tags    time                            value
----    ----    ----                            -----
cpu             2015-01-26T22:01:11.703Z        100
cpu             2015-01-27T22:01:11.703Z        100
cpu             2015-01-28T22:01:11.703Z        100
```

**TO**:

```sql
> select * from cpu
name    time                            value
----    ----                            -----
cpu     2015-01-26T22:01:11.703Z        100
cpu     2015-01-27T22:01:11.703Z        100
cpu     2015-01-28T22:01:11.703Z        100
```

**FROM**:

```sql
> show series
name    tags    id      host
----    ----    --      ----
cpu             1       server01
----    ----    --      ----
mem             2       server01
```

**TO**:

```sql
> show series
name    id      host
----    --      ----
cpu     1       server01
name    id      host
----    --      ----
mem     2       server01
```

**FROM**:

```sql
> show databases
name    tags    name
----    ----    ----
                foo
```

**TO**:

```sql
> show databases
name
----
foo
```

### CODE REVIEW

**Comment by otoolep** [comment]
In the first example, there are no more tags? And how does it look if there is more than 1 value? Can you show that?

**Comment by corylanou** [comment]
with tags:

```sql
> select * from temp group by *
name    tags                            time                            value
----    ----                            ----                            -----
temp    host=server01,region=useast     2015-02-27T22:01:11.703Z        101.1
name    tags                            time                            value
----    ----                            ----                            -----
temp    region=uswest,host=server01     2015-02-26T22:01:11.703Z        98.6
name    tags                            time                            value
----    ----                            ----                            -----
temp    host=server02,region=useast     2015-02-28T22:01:11.703Z        105.4
```

with multiple values:

```sql
> select * from network
name    time                            tx      rx
----    ----                            --      --
network 2015-02-26T22:01:11.703Z        9804    2342
network 2015-02-27T22:01:11.703Z        7930    4324
network 2015-02-28T22:01:11.703Z        8234    2342
```

File: cmd/influx/main.go [hunk 1]
```diff
@@ -320,16 +320,32 @@ func WriteColumns(results *client.Results, w io.Writer) {
...
+			columnNames = append([]string{"tags"}, columnNames...)
+		}
+
+		if row.Name != "" {
+			columnNames = append([]string{"name"}, columnNames...)
+		}
+		for _, column := range row.Columns {
+			columnNames = append(columnNames, column)
 		}
+		// Output a line seperator if we have more than one set or results
```
Comment (otoolep) [inline cmd/influx/main.go]: Nit: 'separator' is misspelled.

File: cmd/influx/main.go [hunk 2]
```diff
@@ -320,16 +320,32 @@ func WriteColumns(results *client.Results, w io.Writer) {
...
+		// gather tags
+		var hasTags bool
+		tags := []string{}
+		for k, v := range row.Tags {
+			hasTags = true
+			tags = append(tags, fmt.Sprintf("%s=%s", k, v))
+		}
+
+		columnNames := []string{}
+		if hasTags {
```
Comment (otoolep) [inline cmd/influx/main.go]: You don't need this variable. Simply check if `len(tags) > 0`, right?

File: cmd/influx/main.go [hunk 3]
```diff
@@ -309,7 +309,7 @@ func WriteColumns(results *client.Results, w io.Writer) {
 		// Create a tabbed writer for each result a they won't always line up
 		w := new(tabwriter.Writer)
 		w.Init(os.Stdout, 0, 8, 1, '\t', 0)
-		csv := resultToCSV(result, "\t", true)
+		csv := resultToCSV(result, "\t", false)
```
Comment (otoolep) [inline cmd/influx/main.go]: This on purpose?

**Comment by otoolep** [comment]
I like the output changes, though I have some minor feedback on the code. Once that is addressed, +1 form me.

**Comment by corylanou** [comment]
Ok, so based on feedback, I have made quite a few changes to the `column` format output.  I left the `csv` format in tact if we need it for future (which I think makes sense as you would be able to use this in a non-interactive mode to get results).

Here are the new examples:

```sql
> show databases
name
foo

> show series
cpu
---
id
1

mem
---
id      host
2       server01
3       server02

network
-------
id      host            region
7       server01        uswest
8       server01        useast
9       server02        useast

temp
----
id      host            region
4       server01        uswest
5       server01        useast
6       server02        useast

> show measurements
measurements
------------
name
cpu
mem
network
temp
> select * from cpu
cpu
---
time                            value
2015-02-26T22:01:11.703Z        8.9
2015-02-27T22:01:11.703Z        1.3
2015-02-28T22:01:11.703Z        50.4

> select * from temp group by *
temp
----------------------------------
tags: host=server01, region=useast
----------------------------------
time                            value
----                            -----
2015-02-27T22:01:11.703Z        101.1

temp
----------------------------------
tags: host=server01, region=uswest
----------------------------------
time                            value
----                            -----
2015-02-26T22:01:11.703Z        98.6

temp
----------------------------------
tags: host=server02, region=useast
----------------------------------
time                            value
----                            -----
2015-02-28T22:01:11.703Z        105.4

```

cc @otoolep @pauldix

**Comment by otoolep** [comment]
Is this still open to feedback? Because I think the output from `select * from temp group by *` is still too hard to read. There are too many lines, which means it's hard to quickly see what is grouped with what. My eyes keep being drawn to what is surrounded by `---` so it looks like `tags` is the section title, not the measurement name. How about the simplified version below?

```
> select * from temp group by *
temp
tags: host=server01, region=useast
time                            value
----                            -----
2015-02-27T22:01:11.703Z        101.1

temp
tags: host=server01, region=uswest
time                            value
----                            -----
2015-02-26T22:01:11.703Z        98.6

temp
tags: host=server02, region=useast
time                            value
----                            -----
2015-02-28T22:01:11.703Z        105.4
```

**Comment by otoolep** [comment]
Or:
```
> select * from temp group by *
measurement: temp
tags: host=server01, region=useast
time                            value
----                            -----
2015-02-27T22:01:11.703Z        101.1

measurement: temp
tags: host=server01, region=uswest
time                            value
----                            -----
2015-02-26T22:01:11.703Z        98.6

measurement: temp
tags: host=server02, region=useast
time                            value
----                            -----
2015-02-28T22:01:11.703Z        105.4
```
This form has the advantage of helping our users understand the term 'measurement', as used within our system.

**Comment by corylanou** [comment]
That format was proposed by @pauldix .  I'll let you guys come to consensus.  FWIW, I think I like less lines per what @otoolep proposes.

As for 

```sql
measurement: temp
```

I don't always now that `row.Name` is a measurement name, as sometimes it isn't.

**Comment by otoolep** [comment]
OK, if `row.Name` is not always a measurement, then scrap that idea.

**Comment by corylanou** [comment]
What if we did:

```sql
> select * from temp group by *
name: temp
tags: host=server01, region=useast
time                            value
----                            -----
2015-02-27T22:01:11.703Z        101.1


name: temp
tags: host=server01, region=uswest
time                            value
----                            -----
2015-02-26T22:01:11.703Z        98.6


name: temp
tags: host=server02, region=useast
time                            value
----                            -----
2015-02-28T22:01:11.703Z        105.4
```

And if we clean up some inconsistencies in responses for things that don't need names, like `show databases` or `show measurements` it will actually work out quite nicely.

**Comment by otoolep** [comment]
Can you show a couple more examples? How does stuff like `count` look?

**Comment by corylanou** [comment]
@otoolep latest outputs:

```sql
> show databases
name
foo

> show series
name: cpu
---------
id
1


name: mem
---------
id      host
2       server01
3       server02


name: network
-------------
id      host            region
7       server01        uswest
8       server01        useast
9       server02        useast


name: temp
----------
id      host            region
4       server01        uswest
5       server01        useast
6       server02        useast

> show measurements
name: measurements
------------------
name
cpu
mem
network
temp

> select * from temp group by *
name: temp
tags: host=server01, region=useast
time                            value
----                            -----
2015-02-27T22:01:11.703Z        101.1


name: temp
tags: host=server01, region=uswest
time                            value
----                            -----
2015-02-26T22:01:11.703Z        98.6


name: temp
tags: host=server02, region=useast
time                            value
----                            -----
2015-02-28T22:01:11.703Z        105.4

> select count(value) from cpu
name: cpu
---------
time                    count
1970-01-01T00:00:00Z    3

```

**Comment by corylanou** [comment]
I think we are as far as we are going to get on this for now.  It fixes several output format bugs, and has improved greatly from what it was.  We can come back later to tweak further at a future date.

**Comment by toddboom** [comment]
@corylanou rad, can you rebase and push? i'll merge it in.

**Comment by toddboom** [comment]
@corylanou nevermind. ;)

**Comment by pauldix** [comment]
@corylanou latest looks rad, :ship: while it's hot!

**Pull request merged**